overlord/snapstate/check_snap.go

```diff
@@ -566,10 +566,21 @@ func checkAndCreateSystemUsernames(si *snap.Info) error {
 	return nil
 }
 
+func checkConfigureHooks(_ *state.State, snapInfo, curInfo *snap.Info, _ snap.Container, _ Flags, deviceCtx DeviceContext) error {
+	hasDefaultConfigureHook := snapInfo.Hooks["default-configure"] != nil
+	hasConfigureHook := snapInfo.Hooks["configure"] != nil
+
+	if hasDefaultConfigureHook && !hasConfigureHook {
+		return fmt.Errorf(`cannot use "default-configure" hook without "configure" hook`)
```

Comment:
s/use/specify/ ?

Reply:
thanks for the change, we need to make sure this issue would also be detected by snap pack and in general snap validation: this can be done in a follow up but we need:

snap.Validate to detect this kind of issue
and also "snap pack --check-skeleton", that means snap/pack/pack.go loadAndValidate,  there is some behavior related to implicit hooks (hooks that exists as file but not mentioned in snap.yaml) - see addImplicitHooks in the snap package - that doesn't seem to be performed during snap pack checks, so this might need some reorg changes about that

Reply:
btw once snap.Validate does this check we might not need this code here

Reply:
Thanks, I changed use to specify - sounds better.

Reply:
Created ticket for additional checking: SNAPDENG-7297